src/schemas/config-1.0.1.schema.json

```diff
@@ -0,0 +1,846 @@
+            }
+          ]
+        },
+        "index": {
+          "type": "number",
+          "description": "The index of the layer among the array of layers available in the image file."
+        },
+        "opacity": {
+          "type": "number"
+        },
+        "modelMatrix": {
+          "oneOf": [
+            {
+              "type": "array",
+              "minItems": 16,
+              "maxItems": 16,
+              "description": "transformation matrix for this layer"
+            },
+            {
+              "type": "null",
+              "description": "Use no transformation."
+            }
+          ]
+        },
+        "domainType": {
+          "type": "string",
+          "enum": ["Full", "Min/Max"],
+          "description": "Determines the extent of the channel slider input element in the layer controller."
+        },
+        "type": {
+          "type": "string",
+          "enum": ["raster"]
+        }
+      }
+    },
+    "moleculesLayer": {
+      "description": "The properties of this object are the rendering settings for the molecules layer.",
+      "additionalProperties": false,
+      "required": ["visible", "radius", "opacity"],
+      "properties": {
+        "visible": {
+          "type": "boolean"
+        },
+        "radius": {
+          "type": "number"
+        },
+        "opacity": {
+          "type": "number"
+        }
+      }
+    },
+    "cellsLayer": {
+      "description": "The properties of this object are the rendering settings for the cells layer.",
+      "additionalProperties": false,
+      "required": ["visible", "stroked", "radius", "opacity"],
+      "properties": {
+        "visible": {
+          "type": "boolean"
+        },
+        "stroked": {
+          "type": "boolean"
+        },
+        "radius": {
+          "type": "number"
+        },
+        "opacity": {
+          "type": "number"
+        }
+      }
+    },
+    "neighborhoodsLayer": {
+      "description": "The properties of this object are the rendering settings for the neighborhoods layer.",
+      "additionalProperties": false,
+      "required": ["visible"],
+      "properties": {
+        "visible": {
+          "type": "boolean"
+        }
+      }
+    },
+    "spatialRasterLayers": {
+      "type": "array",
+      "description": "Array of Spatial Layers",
+      "items": {
+        "$ref": "#/definitions/rasterLayer"
+      }
+    },
+    "spatialCellsLayers": {
+      "type": "array",
+      "description": "Array of Spatial Layers",
+      "items": {
+        "$ref": "#/definitions/cellsLayer"
+      }
+    },
+    "spatialMoleculesLayers": {
+      "type": "array",
+      "description": "Array of Spatial Layers",
+      "items": {
+        "$ref": "#/definitions/moleculesLayer"
+      }
+    },
+    "spatialNeighborhoodsLayers": {
+      "type": "array",
+      "description": "Array of Spatial Layers",
+      "items": {
+        "$ref": "#/definitions/neighborhoodsLayer"
+      }
+    }
```

Comment:
Yeah I am not really sure what the best approach is. I guess it is future-proofing but I don't have a good sense of a future situation for which an array would be needed.

Reply:
Maybe there could be nucleus segmentations and cytoplasm segmentations, and these are two different cell layers that someone wants to see overlaid? At that point I feel like implementing specific layers for those two types of segmentations may be a better solution though

Reply:
Given this, I think switching this schema to restrict to single objects would be better for now.

Especially since single objects would be backwards compatible with an array of objects (upgrading the schema version would just result in an array of length 1). But once we allow an array of objects, it would not really be backwards compatible if we ever wanted to switch to single objects

Reply:
That was what I was thinking in terms of "future-proofing," but maybe it's too complicated for this model, yes

Reply:
> But once we allow an array of objects, it would not really be backwards compatible if we ever wanted to switch to single objects

We actually already kind of have this problem for upgrading the schema version from the current 1.0.0 to the 1.0.1 in this PR where we split up the `spatialLayers` array into the separate `spatialCellsLayers` and `spatialMoleculesLayers` since in theory there is nothing in the 1.0.0 schema stopping from adding two layers with `type: "cell"`. But I think assuming there is only one is fine since having more than one does not make sense in the overall context of Vitessce right now

Reply:
We will need to encode this assumption in the upgrade method here https://github.com/vitessce/vitessce/blob/d5480d9/src/app/view-config-utils.js#L316-L322

Reply:
And then the `spatialRasterLayers` would still be an array?

Reply:
Yes since we still need to support multiple image layers